packages/lazy-fn-vite-plugin/src/index.ts

```diff
@@ -91,28 +105,38 @@ const TurnLazyIntoStaticImport: PluginItem = {
   },
 }
 
-export function LazyLoadingPlugin(): Plugin {
+export function LazyLoadingPlugin(): VitePlugin {
   return {
     name: 'vite-plugin-tuono-lazy-loading',
     enforce: 'pre',
-    transform(code, _id, opts): string | undefined | null {
+    transform(code, _id, opts): Rollup.TransformResult {
+      /**
+       * @todo we should exclude non tsx files from this transformation
+       *       this might benefit build time avoiding running `includes` on non-tsx files.
+       *       This can be executed using `_id` parameter
+       *       which is the filepath that is being processed
+       */
+
```

Comment:
Let me know what do you think about this

Reply:
I agree. We should be extra-careful though. We should support at least js/jsx/mdx

Reply:
New separate issue: #224